R/zchunk_L133.water.demand.livestock.R

```diff
@@ -36,41 +35,128 @@ module_water_L133.water.demand.livestock_DISABLED <- function(command, ...) {
+      filter(year == 2000) ->
+      L133.FAO_an_heads
+
+
+    # Now calculate the water demand by FAO item.
     #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
+    # Add FAO stock information, livestock water use coefficient, and GCAM information to the dairy and non-dairy
+    # livestock count form mapping files.
+    L133.FAO_an_heads %>%
+      # A 1:1 match is not expected and we do not want NAs introduced to the data frame so
+      # use inner join here.
+      inner_join(FAO_an_items_Stocks, by = "item") %>%
+      # A 1:1 match is not expected and we do not want NAs introduced to the data frame so
+      # use inner join here.
+      inner_join(LivestockWaterFootprint_MH2010, by = "Animal") %>%
+      left_join_error_no_match(iso_GCAM_regID, by = "iso") ->
+      L133.FAO_an_heads
+
+    # Multiply the livestock count by the livestock water use coefficient from Mekonnen and Hoekstra 2010.
     #
+    # Since the Mekonnen and Hoekstra 2010 coefficient is in liters/head per day convert from L to m^3 per
+    # 1000 heads by dividing by 1000 and then convert from daily consumption to per year.
+    L133.FAO_an_heads %>%
+      mutate(water.consumption = value * Coefficient / 1000 / CONV_DAYS_YEAR) ->
+      L133.FAO_an_heads
+
+
+    # Calculate water demand by GCAM_commodity
     #
-    # NOTE: there are `merge` calls in this code. Be careful!
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Aggregate the livestock water consumption by GCAM region and commodity.
+    L133.FAO_an_heads %>%
+      group_by(GCAM_region_ID, GCAM_commodity) %>%
+      summarise(water.consumption = sum(water.consumption)) %>%
+      ungroup ->
+      L133.water_demand_livestock_R_C_W_km3_Mt
+
+    # Add FAO production information to the tibble of aggregated livestock water consumption.
+    L133.water_demand_livestock_R_C_W_km3_Mt %>%
+      left_join_error_no_match(L105.an_Prod_Mt_R_C_Y %>%
+                                 filter(year == 2000) %>%
+                                 select(GCAM_region_ID, GCAM_commodity, year, value),
+        by = c("GCAM_region_ID", "GCAM_commodity")) ->
+      L133.water_demand_livestock_R_C_W_km3_Mt
+
+    # Average the aggregated livestock water consumption by the total production. Since water
+    # consumption is in m^3 and production is in Mt to km^3/Mt we must divide by 1e9.
+    L133.water_demand_livestock_R_C_W_km3_Mt %>%
+      mutate(coefficient = water.consumption / value / 1e9) ->
+      L133.water_demand_livestock_R_C_W_km3_Mt
+
+    # Add the water type information to the livestock water demand tibble. Add water type information
```

Comment:
Could you put a one line comment right at the top, "Water withdrawals are assumed to be the same as consumption." so that this is clear. It's a bit buried in the comments and I don't want folks to miss that.